pkg/server/io/container_io.go

```diff
@@ -105,18 +105,20 @@ func (c *ContainerIO) Config() cio.Config {
 // to output stream.
 func (c *ContainerIO) Pipe() {
 	wg := c.closer.wg
```

Comment:
won't hurt to test if c.closer is nil

Reply:
The closer can never be nil.

If it could be nil, we need to test each of the `wg` is nil or not underlying. If so, I prefer just assume that closer won't be nil here. :)

Reply:
never is a long time :) but yeah one would want to add a c.getcloserwg() func that logs a debug if c.closer test == nil and otherwise returns the wg, since this code isn't executed very often it's not a performance issue, vs doing the test 4 times.